#!/usr/bin/env python
PACKAGE = "plane_slam"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

##
gen.add("focus_on_camera", bool_t, 0, "", False)
gen.add("display_frame", bool_t, 0, "", False)
gen.add("display_input_cloud",  bool_t, 0,  "", False)
gen.add("display_line_cloud",  bool_t, 0,  "", False)
gen.add("display_normal",   bool_t, 0,  "", False)
gen.add("display_normal_arrow", bool_t, 0, "", False)
gen.add("display_plane",    bool_t, 0,  "", False)
gen.add("display_plane_number",    bool_t, 0,  "", False)
gen.add("display_plane_arrow",    bool_t, 0,  "", False)
gen.add("display_plane_inlier", bool_t, 0, "", True)
gen.add("display_plane_projected_inlier",    bool_t, 0,  "", False)
gen.add("display_plane_boundary", bool_t, 0, "", True)
gen.add("display_plane_hull",    bool_t, 0,  "", False)
gen.add("display_feature_cloud", bool_t, 0, "", False)
##
gen.add("show_keypoint", bool_t, 0, "", False)
gen.add("show_keypoint_matches", bool_t, 0, "", False)
gen.add("display_3d_keypoint_matches", bool_t, 0, "", False)
##
gen.add("display_plane_landmarks", bool_t, 0, "", False )
gen.add("display_point_landmarks", bool_t, 0, "", False )
gen.add("display_landmark_inlier", bool_t, 0, "", True )
gen.add("display_landmark_arrow", bool_t, 0, "", False )
gen.add("display_landmark_number", bool_t, 0, "", False )
gen.add("display_landmark_boundary", bool_t, 0, "", False )
gen.add("display_landmark_hull", bool_t, 0, "", False )
gen.add("display_landmark_label", bool_t, 0, "", True )
#
gen.add("display_camera_fov", bool_t, 0, "", True )
gen.add("display_pathes", bool_t, 0, "", True )
gen.add("display_optimized_path", bool_t, 0, "", True )
gen.add("display_odom_path", bool_t, 0, "", True )
gen.add("display_visual_odom_path", bool_t, 0, "", True )
gen.add("display_true_path", bool_t, 0, "", True )


exit(gen.generate(PACKAGE, "plane_slam", "Viewer"))
